modules/twinklewarn.js

```diff
@@ -1644,16 +1475,18 @@ Twinkle.warn.callbacks = {
 		text += "|sig=true}}";
 		return text;
 	},
+	// NOTE: used by block module
 	preview: function(form) {
-		var templatename = form.sub_group.value;
-		var linkedarticle = form.article.value;
-		var templatetext;
-
-		if (templatename in Twinkle.warn.messages.block) {
-			templatetext = Twinkle.warn.callbacks.getBlockNoticeWikitext(templatename, linkedarticle, form.block_timer.value,
-				form.block_reason.value, Twinkle.warn.messages.block[templatename].indefinite);
+		var templatename = form.sub_group.value,
+				linkedarticle = form.article.value,
+				templatetext;
+
+		if (templatename in Twinkle.block.blockPresetsInfo) {
+			// form.expiry.value is for block module, where expiry is the key name that the API uses
```

Comment:
I actually don't know what I was trying to say here in the comment. Warn wants `block_timer`,
where with Block we use `expiry` as that's what the API wants. However we also want `block_timer` because you can also use Block to warn (issue block template). They are separate because I wanted to give the ability to provide any duration string value for the template (e.g. "temporarily", "a fortnight", etc.), however I thought it was weird to have two duration input fields in the form. On second thought maybe they should both just be visible.